### PR TITLE
refactor of Snyk × bomber

### PR DESCRIPTION
**Pull request by mcombuechen**
* responsibilities extracted and moved into files:
  * `getOrgID` (userinfo introspection)
  * `getVulnsForPurl` (request to issues endpoint)
  * `snykIssueToBomberVuln` (mapping between schemas)
* extended population of bomber Vulnerability values
* increased test coverage

### TODO

- [x] support for `SNYK_API_TOKEN` env var
- [x] unskip tests in `snyk_test.go`

### CODE REVIEW

File: providers/snyk/client.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+	"github.com/kirinlabs/HttpRequest"
+
+	"github.com/devops-kung-fu/bomber/models"
+)
+
+const USER_AGENT = "Bomber"
+
+func newClient(c *models.Credentials) *HttpRequest.Request {
+	return HttpRequest.NewRequest().SetHeaders(map[string]string{
+		// "Content-Type":  "application/vnd.api+json",
```
Comment (mcombuechen) [inline providers/snyk/client.go]: @garethr I don't believe this was needed for GET requests. What do you think?

File: providers/snyk/vulns.go [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+	return models.Vulnerability{
+		ID:                 v.Id,
+		Title:              v.Attributes.Title,
+		Description:        v.Attributes.Description,
+		Severity:           severity,
+		Cwe:                getCwe(v),
+		CvssScore:          float64(cvss.Score),
+		CvssVector:         cvss.Vector,
+		Reference:          fmt.Sprintf("https://security.snyk.io/vuln/%s", v.Id),
```
Comment (mcombuechen) [inline providers/snyk/vulns.go:205]: @garethr unfortunately we don't include links to security.snyk.io in the Issues response, hence the hacky construction. Maybe a new feature request for the service?

File: providers/snyk/vulns_test.go [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+}
+
+func TestSnykIssueToBomberVulnSnykSeverity(t *testing.T) {
+	tc := []struct {
+		Title             string
+		Severities        []Severity
+		ExpectedCvssScore float64
+	}{
+		{
+			"prefer NVD score",
```
Comment (mcombuechen) [inline providers/snyk/vulns_test.go]: @garethr the code will pick NVD first, then Snyk, then others. Let me know if we should prefer Snyk sev evaluation. 🙏

Comment (garethr) [reply providers/snyk/vulns_test.go]: We should prefer Snyk first.

It's a good conversation to have later with the Bomber team about the nuance around multiple scores/sources/

File: cmd/scan.go [hunk 4]
```diff
@@ -71,7 +71,7 @@ var (
 					util.PrintInfo("Ecosystems detected:", strings.Join(ecosystems, ","))
 					util.PrintInfof("Scanning %v packages for vulnerabilities...\n", len(purls))
 					util.PrintInfo("Vulnerability Provider:", provider.Info(), "\n")
-					s.Suffix = fmt.Sprintf(" Fetching vulnerability data from %s", providerName)
```
Comment (garethr) [inline cmd/scan.go]: This change appears to trigger a bug for me.

I'd suggest keeping the PR just to the Snyk provider, and then opening a separate issue/PR around this if needed.

For me, this triggered the following behaviour.

```
| Fetching vulnerability data from osv
/ Fetching vulnerability data from osv
- Fetching vulnerability data from osv
\ Fetching vulnerability data from osv
| Fetching vulnerability data from osv
/ Fetching vulnerability data from osv
- Fetching vulnerability data from osv
```

File: README.md [hunk 5]
```diff
@@ -27,7 +27,7 @@ There are quite a few SBOM formats available today. ```bomber``` supports the fo
 
 ## Providers
 
-```bomber``` supports multiple sources for vulnerability information. We call these *providers*. Currently, ```bomber``` uses [OSV](https://osv.dev) as the default provider, but you can also use the [Sonatype OSS Index](https://ossindex.sonatype.org). 
+```bomber``` supports multiple sources for vulnerability information. We call these *providers*. Currently, ```bomber``` uses [OSV](https://osv.dev) as the default provider, but you can also use the [Sonatype OSS Index](https://ossindex.sonatype.org) or the [Snyk Vulnerability DB](https://security.snyk.io).
```
Comment (garethr) [inline README.md]: I'd leave README changes out for this moment. DJ is going to split the README, and I'll write updated docs in a follow on PR.

**Review by garethr — CHANGES_REQUESTED** [review]
This looks great. Just tried it out locally. Two very small changes and we can tidy up for submission I think.

Note we'll need to rebase the commits and sign things based on https://github.com/devops-kung-fu/bomber/blob/main/CONTRIBUTING.md#use-gpg-to-sign-your-commits

**Comment by mcombuechen** [comment]
closing this in favour of https://github.com/devops-kung-fu/bomber/pull/69

**Pull request closed without merging**